src/XProfile/VirtoCommerce.XProfile/Commands/CreateUserCommandHandler.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using AutoMapper;
+using MediatR;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.DependencyInjection;
+using VirtoCommerce.Platform.Core.Security;
+
+namespace VirtoCommerce.ExperienceApiModule.XProfile.Commands
+{
+    public class CreateUserCommandHandler : IRequestHandler<CreateUserCommand, IdentityResult>
+    {
+        private readonly IMapper _mapper;
+        private readonly IServiceProvider _services;
+
+        public CreateUserCommandHandler(IMapper mapper, IServiceProvider services)
+        {
+            _mapper = mapper;
+            _services = services;
+        }
+
+        public async Task<IdentityResult> Handle(CreateUserCommand request, CancellationToken cancellationToken)
+        {
+            // UserManager<ApplicationUser> requires scoped service
+            using (var scope = _services.CreateScope())
+            {
+                var _userManager = scope.ServiceProvider.GetRequiredService<UserManager<ApplicationUser>>();
```

Comment:
to address the issue with scoped services use this approach https://github.com/VirtoCommerce/vc-platform/blob/master/src/VirtoCommerce.Platform.Security/Services/UserSearchService.cs#L14